internal/webhook/managedcluster_webhook.go

```diff
@@ -73,6 +73,19 @@ func (v *ManagedClusterValidator) ValidateCreate(ctx context.Context, obj runtim
 		return admission.Warnings{"Failed to validate k8s version compatibility with ServiceTemplates"}, fmt.Errorf("failed to validate k8s compatibility: %v", err)
 	}
 
+	cred, err := v.getManagedClusterCredential(ctx, managedCluster.Namespace, managedCluster.Spec.Credential)
+	if err != nil {
+		return nil, fmt.Errorf("%s: %v", invalidManagedClusterMsg, err)
+	}
+
+	if cred.Status.State != hmcv1alpha1.CredentialReady {
+		return nil, fmt.Errorf("%s: Credential is not Ready", invalidManagedClusterMsg)
+	}
+
+	if err := isCredMatchTemplate(cred, template); err != nil {
+		return nil, fmt.Errorf("%s: %v", invalidManagedClusterMsg, err)
+	}
+
```

Comment:
kind of a nit: may it be a good idea to extract that block into a single method/func to reduce the duplication?